### ISSUE TITLE
tsgo not respecting skipLibCheck

### ISSUE THREAD

**Issue by joshcartme** [microsoft/typescript-go#2226]
## Steps to reproduce

A repro is available in https://github.com/joshcartme/tsgo-repros/tree/main/redeclare-global

## Behavior with `typescript@5.9`

`tsc` produces no output

## Behavior with `tsgo`
`tsgo` produces:

```
src/types.ts:4:20 - error TS2430: Interface 'Locale' incorrectly extends interface 'MLocaleExt'.
  Types of property 'weekdays' are incompatible.
    Type '{ (): string[]; (m: Moment, format?: string): string; }' is not assignable to type 'string[]'.

4   export interface Locale extends MLocaleExt {}
                     ~~~~~~


Found 1 error in src/types.ts:4
```

**Comment by joshcartme** [comment]
I did some searching and this started happening with the Nov 21 release.

**Comment by jakebailey** [comment]
This would have been #2134, because we need to be able to issue the errors regardless of the order in which we check the files (or even when the libs were not checked).

Silencing this with `skipLibCheck` has always been dubious; it's "your" code that is attempting to incorrectly declare something, so if you really want to silence that, you need to `// @ts-expect-error` or something in _your_ code, not silence an error that was being reported in a strange place.

You very well could have seen this error positioning depending on what your editor does before diagnostics are checked, e.g. if you had semantic tokens enabled and it just so happened to run before the lib files were checked.

**Comment by joshcartme** [comment]
@jakebailey makes sense!  We'll either fix our busted types or expect an error.  Thanks
